doc/design/resource_ownership_and_GC.md

```diff
@@ -0,0 +1,38 @@
+# Resource Ownership and GC
+
+## Resource Ownership
+
+An etcd cluster will create and manages etcd pods, services. Those resources like pods/services are owned by the etcd cluster.
+Two etcd cluster owns completely disjoint set of resources, even if they have the same name.
+For example, etcd cluster "A" was deleted and later etcd cluster "A" was created again; we think these are two different clusters.
+Thus, the new one won't be managing the resources from the old one, e.g. left as garbage.
+
+As discussed in https://github.com/coreos/etcd-operator/issues/517,
+we correlate owner (i.e. cluster) and its resources by making use of `ObjectMeta.OwnerReferences` field.
+
+For etcd pod and service, they will have only one owner -- its managing cluster.
+
+## GC
+
+GC issue: https://github.com/coreos/etcd-operator/issues/518
+
+We will talk about two GC strategies we are applying in the following.
+
+### Lazy GC
+
+When creating or deleting a cluster, remove all potential garbage via label convention:
+- remove pods with selector label{"etcd_cluster": $cluster_name, "app": "etcd"}
+- remove services with selector label{"etcd_cluster": $cluster_name}
+
+### Periodic full GC
+
+Every interval (10 mins or so), we find out orphaned pods/svcs.
+
+Initially, our use case would be O(10) clusters, and thus total selected items would be at most O(100).
+
+A simple full scanning algorithm:
+- List pods, services, get the max Resource Version (RV1) from PodList or ServiceList
+- List etcd clusters with ListOptions.ResourceVersion >= RV1.
```

Comment:
this is not needed. the controller should have full knowledge of all running etcd clusters at all the time after initialization.